app/services/invoices/calculate_fees_service.rb

```diff
@@ -199,7 +200,14 @@ def should_create_charge_fees?(subscription)
 
       # NOTE: Charges should not be billed in advance when we are just upgrading to a new
       #       pay_in_advance subscription
-      return false if subscription.plan.pay_in_advance? && subscription.invoices.created_before(invoice).count.zero?
+      # check if a subscription fee already present
+      if subscription.plan.pay_in_advance? && !subscription.in_trial_period? && subscription.invoices.created_before(invoice).count.zero?
```

Comment:
Layout/LineLength: Line is too long. [137/120] (https://rubystyle.guide#max-line-length)